mtest/operators_test.go

```diff
@@ -4,15 +4,204 @@ import (
+	waitRebootCompletion(cluster)
+
+	out, stderr, err := execAtWithInput(host1, ckecliSafe("history", "-n", "2"), "jq", "-s", ".")
+	Expect(err).ShouldNot(HaveOccurred(), "stderr: %s", stderr)
+	var history []*cke.Record
+	err = json.Unmarshal(out, &history)
+	Expect(err).ShouldNot(HaveOccurred())
+	Expect(history).Should(HaveLen(2))
+	Expect(history[0].Operation).To(Equal("rebootDequeue"))
+	Expect(history[1].Operation).To(Equal("reboot"))
+	Expect(history[1].Targets).Should(HaveLen(2))
+
+	By("Reboot operation gives up waiting node startup if deadline is exceeded")
+	previousCommand := cluster.Reboot.Command
+	cluster.Reboot.Command = []string{"sleep", "3600"}
+	_, err = ckecliClusterSet(cluster)
+	// wait for the previous reconciliation to be done
+	time.Sleep(time.Second * 3)
+
+	Expect(err).ShouldNot(HaveOccurred())
+	rebootTargets = node1
+	ckecliWithInput([]byte(rebootTargets), "reboot-queue", "add", "-")
+	waitRebootCompletion(cluster)
+
+	out, stderr, err = execAtWithInput(host1, ckecliSafe("history", "-n", "2"), "jq", "-s", ".")
+	Expect(err).ShouldNot(HaveOccurred(), "stderr: %s", stderr)
+	err = json.Unmarshal(out, &history)
+	Expect(err).ShouldNot(HaveOccurred())
+	Expect(history).Should(HaveLen(2))
+	Expect(history[0].Operation).To(Equal("rebootDequeue"))
+	Expect(history[1].Operation).To(Equal("reboot"))
+	Expect(history[1].Info).Should(HavePrefix("failed to reboot some node"))
+
+	cluster.Reboot.Command = previousCommand
+	_, err = ckecliClusterSet(cluster)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	By("Preparing a deployment to test protected_namespaces")
+	rd, err := ioutil.ReadFile(rebootYAMLPath)
+	Expect(err).ShouldNot(HaveOccurred())
+	_, stderr, err = kubectlWithInput(rd, "apply", "-f", "-")
+	Expect(err).ShouldNot(HaveOccurred(), "stderr: %s", stderr)
+
+	psp, err := ioutil.ReadFile(policyYAMLPath)
+	Expect(err).ShouldNot(HaveOccurred())
+	_, stderr, err = kubectlWithInput(psp, "apply", "-f", "-", "-n=reboot-sample")
+	Expect(err).ShouldNot(HaveOccurred(), "stderr: %s", stderr)
+
+	Eventually(func() error {
+		out, _, err = kubectl("get", "-n=reboot-sample", "deployments/sample", "-o=json")
+		if err != nil {
+			return err
+		}
+
+		var deploy appsv1.Deployment
+		err = json.Unmarshal(out, &deploy)
+		if err != nil {
+			return err
+		}
+		if deploy.Status.ReadyReplicas != 3 {
+			return fmt.Errorf("deployment is not ready")
+		}
+		return nil
+	}).Should(Succeed())
+
+	out, _, err = kubectl("get", "-n=reboot-sample", "pod", "-l=reboot-app=sample", "-o=json")
+	Expect(err).ShouldNot(HaveOccurred(), "stderr: %s", stderr)
+	var pods corev1.PodList
+	err = json.Unmarshal(out, &pods)
+	Expect(err).ShouldNot(HaveOccurred())
+	Expect(pods.Items).Should(HaveLen(3))
+
+	By("Reboot operation will protect all pods if protected_namespaces is nil")
+	nodeName := pods.Items[0].Spec.NodeName
+	_, _, err = ckecliWithInput([]byte(nodeName), "reboot-queue", "add", "-")
+	Expect(err).ShouldNot(HaveOccurred())
+
+	Eventually(func() error {
```

Comment:
The queue entry is expected not to be processed in this case.
Why don't you use `Consistently` to check that?